server/src/Irony.cpp

```diff
@@ -286,6 +286,50 @@ void Irony::complete(const std::string &file,
   std::cout << "(success . t)\n";
 }
 
+namespace {
+
+static bool hasUppercase(const std::string &prefix)
+{
+  for (auto it = prefix.begin(); it != prefix.end(); ++it) {
+    if (std::isupper(*it)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+static bool startsWith(const std::string& str, const std::string &prefix, bool caseSensitive)
+{
+  if (str.length() < prefix.length()) {
+    return false;
+  }
+
+  auto charCmp = caseSensitive
+    ? [] (char a, char b) { return a == b; }
+    : [] (char a, char b) { return std::tolower(a) == std::tolower(b); };
+
+  auto res = std::mismatch(prefix.begin(), prefix.end(), str.begin(), charCmp);
+  if (res.first != prefix.end()) {
+    return false;
+  }
+  return true;
+}
+
+static bool isCaseSensitive(const std::string &prefix, std::string caseStyle)
+{
+  if (caseStyle == "" || caseStyle == "nil") {
```

Comment:
I think it would make more sense to check "is caseStyle case-insensitive? Yes, then return true.", and default to case sensitive (function ends with `return false`).